### ISSUE TITLE
Intermediate supervision

### ISSUE THREAD

**Issue by dluvizon** [junhong-3dv/s2m2#27]
Hi, thanks for sharing this amazing work!

I have a question about training.

You explained how the loss function works and gave some other details in the comments. But I could not find whether you supervise only the final model outputs, or if you also have intermediate supervisions.

For instance, in FoundationStereo, they supervise the initial disparity from the cost-volume and the intermediate disparity after each GRU refinement.
In S2M2, do you supervise the initial disparity from OT? What about the intermedia disparities from the refinement blocks?

If you don't supervise the initial disparity from OT, I was wondering how the gradient backprops through it, as I see a `disp.detach()` here: https://github.com/junhong-3dv/s2m2/blob/main/src/s2m2/core/model/s2m2.py#L159

Thanks!
